src/pip/_internal/cli/progress_bars.py

```diff
@@ -17,15 +23,17 @@
 from pip._internal.utils.logging import get_indentation
 
 DownloadProgressRenderer = Callable[[Iterable[bytes]], Iterator[bytes]]
+RichBarType = Literal["auto", "on"]
+BarType = Union[RichBarType, Literal["off"]]
```

Comment:
Not really related to this change, but making BarType an enum would make validation much easier.

Reply:
but it would be tricky to use enum for argparse/optparse